lib/cartopy/mpl/geoaxes.py

```diff
@@ -392,23 +392,27 @@ class GeoAxes(matplotlib.axes.Axes):
     """
     name = 'cartopy.geoaxes'
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, projection=None, **kwargs):
         """
         Create a GeoAxes object using standard matplotlib
         :class:`~matplotlib.axes.Axes` args and kwargs.
 
         Parameters
         ----------
-        map_projection: optional
-            The target :class:`~cartopy.crs.Projection` of this Axes object.
-
-
-        All other args and keywords are passed through to
-        :class:`matplotlib.axes.Axes`.
-
-        """
-        self.projection = kwargs.pop('map_projection')
-        """The :class:`cartopy.crs.Projection` of this GeoAxes."""
+        projection : cartopy.crs.Projection, optional
+            The target projection of this Axes.
+        """
+        if "map_projection" in kwargs:
+            warnings.warn("The `map_projection` keyword argument is "
+                          "deprecated, use `projection` to instantiate a "
+                          "GeoAxes instead.")
+            projection = kwargs.pop("map_projection")
+
+        # The :class:`cartopy.crs.Projection` of this GeoAxes.
+        if not isinstance(projection, ccrs.Projection):
```

Comment:
So I missed this last time. If `projection` is left as its default of `None`, this check fails. Previously, it would have failed on `kwargs.pop('map_projection')`. Was there any reason you made it "optional"?

Reply:
Oh good catch. I can't say I gave this a ton of thought... I was more hoping for consistency in downstream calls like AxesGrid, rather than looking at the specifics.

I guess this should actually be `required` above. It was previously required as you say, but gave a different error if it wasn't included... I'm inclined to update the docstring to `required` and not change it to raising a `KeyError`.

Reply:
Fine with me. Just make sure to also remove the default to `None`.

Reply:
actually, I have no idea why I did that way in the first place... Let's just pop the new keyword in the else branch and then the same error will be thrown. But, we also have the additional check later on to see if it is a Projection.